### ISSUE TITLE
parent-join-client should not depend on JTS and Spatial4J

### ISSUE THREAD

**Issue by dadoonet** [elastic/elasticsearch#28899]
I'm using the High Level Rest Client 6.2.2 which depends on `parent-join-client`.
But this module depends on JTS and Spatial4J:

```xml
<dependency>
  <groupId>org.locationtech.spatial4j</groupId>
  <artifactId>spatial4j</artifactId>
  <version>0.6</version>
  <scope>compile</scope>
</dependency>
<dependency>
  <groupId>com.vividsolutions</groupId>
  <artifactId>jts</artifactId>
  <version>1.13</version>
  <scope>compile</scope>
</dependency>
```

Source: https://search.maven.org/remotecontent?filepath=org/elasticsearch/plugin/parent-join-client/6.2.2/parent-join-client-6.2.2.pom

I'm unsure if it is on purpose or a bug. May be we should mark it at least as optional?
Might be related to #26850.

**Comment by javanna** [comment]
cc @elastic/es-core-infra

**Comment by jasontedor** [comment]
I am not sure what there is to do here? The parent-join module depends on server, as do all modules; the only solutions are to remove jts and spatial4j from server yet I do not consider these core issues. Closing.

**Comment by tvernum** [comment]
@jasontedor 
The spatial4j and JTS dependencies are marked as `optional` in `elasticsearch.pom`, but not in `parent-join-client`.
I think that's just an oversight in the parent-join plugin.

https://search.maven.org/remotecontent?filepath=org/elasticsearch/elasticsearch/6.2.2/elasticsearch-6.2.2.pom
```
<dependency>
  <groupId>org.locationtech.spatial4j</groupId>
  <artifactId>spatial4j</artifactId>
  <version>0.6</version>
  <scope>compile</scope>
  <optional>true</optional>
</dependency>
<dependency>
  <groupId>com.vividsolutions</groupId>
  <artifactId>jts</artifactId>
  <version>1.13</version>
  <scope>compile</scope>
  <optional>true</optional>
</dependency>
```

**Comment by rjernst** [comment]
This is due to PluginBuildPlugin.configureDependencies. clientJar creation currently just copies the server module jar. We will likely need to add a hack for this in addClientJarPomGeneration manipulate the pom directly and convert these back to optional. Alternatively, we could try just removing these from configureDependencies. I don't recall what errors I saw originally, but that was long before we added the client jars, and any module/plugin that does depend on them being present can "upgrade" them to a compileOnly dep inside that module's build.gradle. Finally, note that I think this likely applies to the other things we mark compileOnly there, like log4j and jna....
